core/download.ts

```diff
@@ -40,6 +40,12 @@ function detectType(text: string): string | undefined {
   }
 }
 
+function detectEncodeType(response: Partial<Response>): string {
+  let headers = response.headers ?? new Headers()
+  let contentType = headers.get('content-type')?.toLowerCase() ?? ''
+  return contentType.match(/charset=([a-zA-Z0-9-]+)/)?.[1] ?? 'utf-8'
```

Comment:
```suggestion
  return contentType.match(/charset=(\w+)/)?.[1] ?? 'utf-8'
```

Can we do like this?

Reply:
i think, no. in this case we get `utf` instead of `utf-8`

Reply:
You right. What about `[\w-]+`?

Reply:
We also need `_`

Here is the full list of variants
https://www.iana.org/assignments/character-sets/character-sets.xhtml

Reply:
maybe better include all symbols except space? something like that?
`.match(/charset=([^\s]+)/)`

Reply:
`.match(/;\s*charset=([^\s;]+)/)`

Technically it could be like `Content-Type: text/html;charset=utf-8;boundary=ExampleBoundaryString`

Reply:
oh, you are right. thx